File: ios/ClientQuery.swift
//
//  ClientQuery.swift
//  RNShopifyBuy
//
//  Created by Jérémy Magrin on 2021-02-16.
//  Copyright © 2021 Facebook. All rights reserved.
//

import MobileBuySDK

final class ClientQuery {
    static func queryForShopName() -> Storefront.QueryRootQuery {
        return Storefront.buildQuery { $0
            .shop { $0
                .name()
            }
        }
    }
    
    static func queryForCardVaultUrl() -> Storefront.QueryRootQuery {
        return Storefront.buildQuery { $0
            .shop { $0
                .paymentSettings { $0
                    .cardVaultUrl()
                }
            }
        }
    }
    
    static func queryForCheckout(id: GraphQL.ID) -> Storefront.QueryRootQuery {
        return Storefront.buildQuery { $0
            .node(id: id) { $0
                .id()
                .onCheckout { $0
                    .fragmentForCheckout()
                }
            }
        }
    }
}

File: android/src/main/java/com/rnshopifybuy/core/ClientQuery.kt
package com.rnshopifybuy.core

import com.shopify.buy3.Storefront.*
import com.shopify.graphql.support.ID
import com.rnshopifybuy.fragment.CheckoutFragment

class ClientQuery {
    companion object {
        fun queryForShopName(): QueryRootQuery {
            return query { rootQuery ->
                rootQuery
                        .shop { shopQuery ->
                            shopQuery
                                    .name()
                        }
            }
        }

        fun queryForCardVaultUrl(): QueryRootQuery {
            return query { rootQuery ->
                rootQuery
                        .shop { shopQuery ->
                            shopQuery
                                    .paymentSettings { paymentSettings ->
                                        paymentSettings
                                                .cardVaultUrl()
                                    }
                        }
            }
        }

        fun queryForCheckout(id: ID): QueryRootQuery {
            return query { rootQuery ->
                rootQuery
                        .node(id) { nodeQuery ->
                            nodeQuery
                                    .id()
                                    .onCheckout { checkout ->
                                        CheckoutFragment(checkout).fragment()
                                    }
                        }
            }
        }
    }
}